tsfile/src/main/java/org/apache/iotdb/tsfile/tool/upgrade/UpgradeTool.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iotdb.tsfile.tool.upgrade;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import org.apache.commons.io.FileUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class UpgradeTool {
+
+  private static final Logger logger = LoggerFactory.getLogger(UpgradeTool.class);
+
+  /**
+   * upgrade all tsfiles in the specific dir
+   *
+   * @param dir tsfile dir which needs to be upgraded
+   * @param upgradeDir tsfile dir after upgraded
+   */
+  public static void updateTsfiles(String dir, String upgradeDir) throws IOException {
+    //Traverse to find all tsfiles
+    File file = new File(dir);
+    List<File> tmp = new ArrayList<>();
```

Comment:
Use a heap or stack or at least remove the last element.

Reply:
I changed to use a queue to maintain the structure.